backend/routes/friends.js

```diff
@@ -0,0 +1,79 @@
+const Parse = require('parse/node');
+const express = require("express")
+const router = express.Router()
+var cors = require('cors');
+
+router.use(cors())
+
+router.get('/users/:name', async (req, res) => {
+    try {
+        let query = new Parse.Query("_User")
+        query.equalTo("username", req.params.name)
+        const user = await query.first({useMasterKey : true})
+        res.status(200).send(user)
+    } catch (err) {
+        res.status(400).send({"error" : err })
+    }
+})
+
+router.post('/send/:sessionToken', async (req, res) => {
+    try {
+        let query = new Parse.Query("_Session")
+        query.equalTo("sessionToken", req.params.sessionToken)
+        const session = await query.first({useMasterKey : true})
+        let userQuery = new Parse.Query("_User")
+        userQuery.equalTo("objectId", session.attributes.user.id)
+        let user = await userQuery.first({useMasterKey : true})
+        const Friend = Parse.Object.extend("Friends")
+        let friend = new Friend()
+        friend.set("fromUser", session.attributes.user.id)
+        friend.set("toUser", req.body.friend.objectId)
+        friend.set("fromName", user.attributes.username)
+        friend.set("toName", req.body.friend.username)
+        friend.set("status", "pending")
+        await friend.save()
+        res.status(200).send(friend)
+    } catch (err) {
```

Comment:
again, curious in what cases the try block would fail here.

Reply:
I usually put it there just in case the server or backend database is down, but there's no actual reason it should fail.